packages/@aws-cdk/pipelines/lib/main/pipeline.ts

```diff
@@ -0,0 +1,127 @@
+import { Construct } from 'constructs';
+import { AddStageOpts, AddWaveOptions, Blueprint, Step } from '../blueprint';
+import { IDeploymentEngine } from './engine';
+
+// v2 - keep this import as a separate section to reduce merge conflict when forward merging with the v2 branch.
+// eslint-disable-next-line
+import { Aspects, Construct as CoreConstruct, Stage } from '@aws-cdk/core';
+
+/**
+ * Properties for a `Pipeline`
+ */
+export interface PipelineProps {
+  /**
+   * The build step that produces the CDK Cloud Assembly
+   *
+   * The primary output of this step needs to be the `cdk.out` directory
+   * generated by the `cdk synth` command.
+   *
+   * If you use `SynthStep`, the output directory will automatically be
+   * configured correctly; if you use a different kind of `Step`, you
+   * may need to configure it explicitly.
+   */
+  readonly synthStep: Step;
+
+  /**
+   * The deployment engine
+   *
+   * This controls what deployment system will be used to deploy this
+   * pipeline.
+   *
+   * Right now, only `new CodePipelineEngine()` is supported in this library,
+   * though 3rd implementations may be available.
+   *
+   * The engine is invoked when `buildPipeline()` is called, or when
+   * `app.synth()` is called (whichever happens first).
+   */
+  readonly engine: IDeploymentEngine;
+}
+
+/**
+ * A generic CDK Pipelines pipeline
+ *
+ * The Pipeline should receive an Engine which builds actual infrastructure from
+ * the pipeline definition.
+ *
+ * The actual pipeline infrastructure is constructed (by invoking the engine)
+ * when `buildPipeline()` is called, or when `app.synth()` is called (whichever
+ * happens first).
+ */
+export class Pipeline extends CoreConstruct {
```

Comment:
I am not sure I understand the value of this additional layer. Can't we just merge `Pipeline` and `Blueprint` into a single `Pipeline` class? 

What value is gained by this additional cognitive and maintenance load?